From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cheng Zhao <zcbenz@gmail.com>
Date: Wed, 31 Jan 2024 13:07:31 +0900
Subject: Allow change the root of ffmpeg

https://chromium-review.googlesource.com/c/chromium/third_party/ffmpeg/+/5252191

Some embedders build their own brandings of ffmpeg, for example VS Code
uses a config that disables most codecs, Teams only enables the codecs
that they are paying royalties for.

By adding a ffmpeg_root arg that specifies where to look for code
sources, embedders can put their ffmpeg files in a separate dir instead
of patching the third_party/ffmpeg.

Note: there is a sibling patch with the same name.
One patches the ffmpeg repo, the other patches the chromium repo.

diff --git a/media/ffmpeg/scripts/generate_gn.py b/media/ffmpeg/scripts/generate_gn.py
index a3349aca97d364baa08b8efca2daf0660b0b8a5d..0d5ab90845fcbc1bfea3c1be505ce921c2a5a6ba 100755
--- a/media/ffmpeg/scripts/generate_gn.py
+++ b/media/ffmpeg/scripts/generate_gn.py
@@ -48,7 +48,7 @@ COPYRIGHT = """# Copyright %d The Chromium Authors. All rights reserved.
 """ % (datetime.datetime.now().year)
 
 GN_HEADER = """import("//build/config/arm.gni")
-import("ffmpeg_options.gni")
+import("//third_party/ffmpeg/ffmpeg_options.gni")
 
 # Declare empty versions of each variable for easier +=ing later.
 ffmpeg_c_sources = []
